manifests/pulp/node.pp

```diff
@@ -0,0 +1,105 @@
+# The pulp node configuration
+#
+# In this scenario we assume there is another pulp instance somewhere and this
+# talks to a qpid router somewhere.
+class foreman_proxy_content::pulp::node (
+  String $foreman_url = $::foreman_proxy_content::foreman_url,
+  Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  String $pulp_admin_password = $::foreman_proxy_content::pulp_admin_password,
+  Boolean $qpid_router = $::foreman_proxy_content::qpid_router,
+  String $qpid_router_hub_host = $::foreman_proxy_content::parent_fqdn,
+  Integer[0, 65535] $qpid_router_hub_port = $::foreman_proxy_content::qpid_router_hub_port,
+  String $qpid_router_broker_addr = $::foreman_proxy_content::qpid_router_broker_addr,
+  Integer[0, 65535] $qpid_router_broker_port = $::foreman_proxy_content::qpid_router_broker_port,
+  Boolean $enable_ostree = $::foreman_proxy_content::enable_ostree,
+) {
+  if $qpid_router {
+    include ::foreman_proxy_content::dispatch_router
+
+    qpid::router::connector { 'hub':
+      host         => $qpid_router_hub_host,
+      port         => $qpid_router_hub_port,
+      ssl_profile  => 'client',
+      role         => 'inter-router',
+      idle_timeout => 0,
+    }
+
+    qpid::router::link_route { 'hub-pulp-route-in':
+      prefix    => 'pulp.',
+      direction => 'in',
+    }
+
+    qpid::router::link_route { 'hub-pulp-route-out':
+      prefix    => 'pulp.',
+      direction => 'out',
+    }
+
+    qpid::router::link_route { 'hub-qmf-route-in':
+      prefix    => 'qmf.',
+      direction => 'in',
+    }
+
+    qpid::router::link_route { 'hub-qmf-route-out':
+      prefix    => 'qmf.',
+      direction => 'out',
+    }
+  }
+
+  include ::apache
+  include ::apache::mod::proxy
+  include ::apache::mod::proxy_http
+
+  file {'/etc/httpd/conf.d/pulp_nodes.conf':
+    ensure  => file,
+    content => template('foreman_proxy_content/pulp_nodes.conf.erb'),
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0644',
+  }
+
+  include ::foreman_proxy_content::pub_dir
+
+  include ::certs::qpid
+  class { '::qpid':
+    ssl                    => true,
+    ssl_cert_db            => $::certs::nss_db_dir,
+    ssl_cert_password_file => $::certs::qpid::nss_db_password_file,
+    ssl_cert_name          => 'broker',
+    interface              => 'lo',
+    subscribe              => Class['certs', 'certs::qpid'],
+  }
+
+  include ::certs::apache
+  include ::certs::qpid_client
+  class { '::pulp':
```

Comment:
Is it worth considering centralizing Pulp install and configuration in one place at this juncture? I don't want to over complicate this large re-factor but as we've pointed out in the past we setup Pulp here and in puppet-katello.

Reply:
I considered moving it all into puppet-katello, but you're right we have overlap. We should at least centralize pulp in one place.